ledger-core/v2/virtual/execute/execute.go

```diff
@@ -440,6 +474,9 @@ func (s *SMExecute) setNewState(prototype reference.Global, memory []byte) func(
 			memory,
 			parentReference,
 		))
+		if !s.migrationHappened {
```

Comment:
Why we shouldn't decrement here?

Reply:
We just don't need them anymore, since pulse was changed

Reply:
Then it's just redundant if, that may lead to confusion, imho

Reply:
Possible, yes, It's better to decrement in any case

Reply:
as I wrote in other comment, decrementing in either case result in a bug

Reply:
Right now, don't change counters if pulse was changed.
In the future, when VStateReport get state from immutable SM, we will just remove this 'if' statement